### PR TITLE
[fix #3593] Hide passphrase field from TestFairy

### LINKED ISSUE #3388: Upgrade TestFairy
[comment]: # (Please replace ... with your information. Remove < and >)

### User Story

As a PM, I want to submit app to App Store.

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Bug

[comment]: # (Describe the feature you would like, or briefly summarise the bug and what you did, what you expected to happen, and what actually happens. Sections below)
*Summary*: App uses non-public APIs, which is not permitted on the App Store.

#### Expected behavior
App accepted by Apple.

#### Actual behavior
App not accepted by Apple.

### Reproduction
[comment]: # (Describe how we can replicate the bug step by step.)

Submit app to Apple for release.

To reproduce symptom:

```
find . -type f -exec grep -H 'SCRCException' {} \; Binary file ./ios/libTestFairy.a matches
```

### Solution
[comment]: # (Please summarise the solution and provide a task list on what needs to be fixed.)
*Summary*: 

- [ ] Upgrade TestFairy
- [ ] Make sure app still runs in release mode

Alternatively:

- [ ] Release build script that deletes the Testfairy files (suggested by @rasom as quickfix).

### Additional Information
* Status version: all
* Operating System: iOS

During an app review Apple flagged the following problem:

> Guideline 2.5.1 - Performance - Software Requirements
> 
> Your app uses or references the following non-public APIs:
> 
> PrivateFrameworks/ScreenReaderCore.framework (SCRCException)
> 
> The use of non-public APIs is not permitted on the App Store because it can lead to a poor user experience should these APIs change. 
> 
> Continuing to use or conceal non-public APIs in future submissions of this app may result in the termination of your Apple Developer account, as well as removal of all associated apps from the App Store.
> 
> Next Steps
> 
> If you are using third-party libraries, please update to the most recent version of those libraries. If you do not have access to the libraries' source, you may be able to search the compiled binary using the "strings" or "otool" command line tools. The "strings" tool can output a list of the methods that the library calls and "otool -ov" will output the Objective-C class structures and their defined methods. These tools can help you narrow down where the problematic code resides. You could also use the "nm" tool to verify if any third-party libraries are calling these APIs.
> 
> Resources
> 
> For information on the "nm" tool, please review the "nm tool" Xcode manual page.

#### Logs

### LINKED ISSUE #3593: Hide passphrase field when recording TestFairy session
### Description

*Type*: Feature

*Summary*: In TestFairy session (enabled on develop, nightly and PR builds) we need to hide passphrase field, so its content is not shown on video recorded by TestFairy.

How to do this: https://docs.testfairy.com/Security/Hiding_Sensitive_Data.html
 

### Reproduction

- Install nightly build
- use wifi and launch Status app
- On screen Sign in to Status tap on button Other accounts
- tap on button Add existing account
- in Passphrase field type some words
- open TestFairy and check the recorded session: on video words that you've typed in Passphrase field should be not visible 

### Additional Information
* Status version: nightly build March 16
* Operating System: Android and iOS

### PR DESCRIPTION
**Pull request by dmitryn**
Fixes #3388 #3593 

### Summary 

* Hide passphrase field during TestFairy sessions.
* Switch to react-native version of the library

- [ ] Check if it resolves https://github.com/status-im/status-react/issues/3388 after app submission to Apple

status: ready

### CODE REVIEW

**Review by flexsurfer — CHANGES_REQUESTED** [review]
should be done for TestFairy session only  https://docs.testfairy.com/Security/Hiding_Sensitive_Data.html

**Comment by dmitryn** [comment]
@flexsurfer you think hiding passphrase the same way as password with ****** is a bad idea? How passphrase is different from password? Seems like sensitive information to me so i don't want show that to public.

**Comment by dmitryn** [comment]
@denis-sharypin what do you think of it in terms of UX?

**Comment by flexsurfer** [comment]
just try to enter 12 words :)

**Comment by dmitryn** [comment]
@flexsurfer i see what you mean, that would be hard for sure :)
In terms of UX it's not a good solution, but more secure.

**Comment by flexsurfer** [comment]
password is everyday operation, account recovery very rare and secret operation you shouldn't do it in public place

**Comment by dmitryn** [comment]
>password is everyday operation, account recovery very rare and secret operation you shouldn't do it in public place

Makes sense.

I will try to use TestFairy API then, seems like it requires switch to react-native version of testfairy library.

**Comment by denis-sharypin** [comment]
agree with @flexsurfer

**Comment by dmitryn** [comment]
@flexsurfer removed secure-text-entry attribute and made field hidden using TestFairy API.

File: package.json [hunk 1]
```diff
@@ -70,6 +70,7 @@
     "react-native-splash-screen": "3.0.6",
     "react-native-svg": "6.0.1-rc.3",
     "react-native-tcp": "3.3.0",
+    "react-native-testfairy": "^2.8.0",
```
Comment (jeluard) [inline package.json]: We want fixed versions

File: src/status_im/ui/screens/accounts/recover/views.cljs [hunk 2]
```diff
@@ -9,20 +10,24 @@
...
     [text-input/text-input-with-label
-     {:style          {:flex 1}
-      :height         92
-      :label          (i18n/label :t/passphrase)
-      :placeholder    (i18n/label :t/enter-12-words)
-      :multiline      true
-      :default-value  passphrase
-      :on-change-text #(re-frame/dispatch [:set-in [:accounts/recover :passphrase] %])
-      :error          error}]))
+     {:style               {:flex 1}
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/recover/views.cljs]: Use components/flex

File: src/status_im/ui/screens/accounts/recover/views.cljs [hunk 3]
```diff
@@ -9,20 +10,24 @@
...
             [status-im.ui.screens.accounts.db :as db]
+            [status-im.react-native.js-dependencies :as js-dependencies]
             [cljs.spec.alpha :as spec]
             [status-im.ui.components.common.common :as components.common]))
 
 (defview passphrase-input [passphrase]
-  (letsubs [error [:get-in [:accounts/recover :passphrase-error]]]
+  (letsubs [error [:get-in [:accounts/recover :passphrase-error]]
+            input-ref (reagent/atom nil)]
+    {:component-did-mount #(.hideView js-dependencies/testfairy @input-ref)}
```
Comment (jeluard) [inline src/status_im/ui/screens/accounts/recover/views.cljs]: Only if TF is enabled?

**Review by jeluard — COMMENTED** [review]
Do we want to migrate to a new dependency? Is it not possible to do the same with the old model?

Also is TF always bundled in the app? I understand it wasn't for release build.

File: package.json (hunk 1 above)
Comment (dmitryn) [reply package.json]: Will change to  2.8.0 fixed

File: src/status_im/ui/screens/accounts/recover/views.cljs (hunk 2 above)
Comment (dmitryn) [reply src/status_im/ui/screens/accounts/recover/views.cljs]: Sorry, but seems like not in scope of this PR, i didn't change that line.

**Comment by dmitryn** [comment]
>Do we want to migrate to a new dependency? Is it not possible to do the same with the old model?

@jeluard had to migrate to RN version of TF, otherwise we need to implement Bridge which TF guys actually did in RN version of the lib.

**Comment by dmitryn** [comment]
>Also is TF always bundled in the app? I understand it wasn't for release build.

@jeluard i'm not sure how to exclude that lib from release version.

File: src/status_im/ui/screens/accounts/recover/views.cljs (hunk 2 above)
Comment (jeluard) [reply src/status_im/ui/screens/accounts/recover/views.cljs]: Right but that one of those easy win we can make along the line

**Comment by jeluard** [comment]
@dmitryn Right, just wondering if we loose something with this change. Maybe @rasom can give us some insights?

**Comment by dmitryn** [comment]
@flexsurfer could you review again?

**Comment by dmitryn** [comment]
> I understand it wasn't for release build.

@jeluard yes, TF is disabled in release build.

**Review by jeluard — APPROVED** [review]

**Review by flexsurfer — APPROVED** [review]

**Comment by asemiankevich** [comment]
@dmitryn i can see it is not hidden  on Android
https://app.testfairy.com/projects/4803622-status/builds/7987172/sessions/1/?accessToken=pmrQATgdsWSHpSP3sH57goBL0es

but is hidden on iOS https://app.testfairy.com/projects/4803590-status/builds/7994876/sessions/8/?accessToken=vzsp1SgjJfH3D/z8D-VL9TgTLcc

also i was thinking about hiding it when user goes though the Back up Seed Phrase flow, @annadanchenko WDYT?

image


https://app.testfairy.com/projects/4803622-status/builds/7987172/sessions/2/?accessToken=BrzudhqtIxZq0zYjSS7GLmNItQU

https://app.testfairy.com/projects/4803590-status/builds/7994876/sessions/9/?accessToken=4glWiT8jBX1Zh1o3DaS6W-1cWcQ

**Comment by dmitryn** [comment]
>also i was thinking about hiding it when user goes though the Back up Seed Phrase flow

@asemiankevich i will hide this as well.

**Comment by dmitryn** [comment]
@asemiankevich i've hidden backup seed screen and made a fix for Android. Please take a look.

**Comment by asemiankevich** [comment]
@dmitryn on it

**Comment by asemiankevich** [comment]
1. Seed phrase is not shown on android in TF when recovering account https://app.testfairy.com/projects/4803622-status/builds/8002721/sessions/1/?accessToken=06K7WtBsDUqtyJFfX7R3Lgjfrng

2. Seed phrase is not shown on iOS in TF when recovering account https://app.testfairy.com/projects/4803590-status/builds/8002722/sessions/1/?accessToken=uAewrl6E-qlvJ2YwbHkYe4NATjA

3. on both platforms i dont see the seed phrase generated when i created a new account , all 12 words are hidden. When i proceed with step 2 and 3 the input is visible , but i assume it is okay as the whole phrase was never visible for Test Fairy

image

image

image

image



<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803622-status/builds/8002721/sessions/1/?accessToken=06K7WtBsDUqtyJFfX7R3Lgjfrng">TestFairy: Status 1e3719d8+ (2376) @1</a></strong></div></blockquote>
<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8002722/sessions/1/?accessToken=uAewrl6E-qlvJ2YwbHkYe4NATjA">TestFairy: Status 0.0.2376 @1</a></strong></div></blockquote>

**Pull request merged**